lib/JSON/Validator.pm

```diff
@@ -519,9 +519,8 @@ sub _validate {
   # Test base schema before allOf, anyOf or oneOf
   if (ref $type eq 'ARRAY') {
     push @{$self->{temp_schema}}, [map { +{%$schema, type => $_} } @$type];
-    push @errors,
-      $self->_validate_any_of($to_json ? $$to_json : $_[1],
-      $path, $self->{temp_schema}[-1]);
+    return $self->_validate_any_of_types($to_json ? $$to_json : $_[1],
+      $path, $self->{temp_schema}[-1], 0);
```

Comment:
Extra ", 0" at the end.

Reply:
let's look at combining them afterwards, to keep the changes here simpler. I don't think it will help any (the subs are all pretty short now, but subtly different in how they manage the loop - one of them short-circuits and the others don't; one keeps track of passing tests and the others don't..), so combining them might make readability worse.